app/page/info/tokens.tsx

```diff
@@ -0,0 +1,193 @@
+  const toggleAccountModal = () => setIsTokenModalOpened(!isTokenModalOpened)
+
+  return (
+    <div>
+      <div className="flex-col justify-center items-center h-auto p-3 border-0 rounded-lg text-base">
+        Tokens:
+        {tokens &&
+          tokens.map((token, index) => {
+            return (
+              <div key={index}>
+                <a
+                  href={`${explorerUrl}token/${token.address}?a=${account.address}`}
+                  target="_blank">
+                  {token.symbol}
+                </a>{" "}
+                {parseFloat(
+                  ethers.formatUnits(
+                    ethers.toBeHex(token.balance),
+                    ethers.toNumber(token.decimals)
+                  )
+                ).toFixed(6)}
+              </div>
+            )
+          })}
+      </div>
+      <div className="col-span-3 cursor-pointer" onClick={toggleAccountModal}>
+        <span>Import Tokens</span>
+        <FontAwesomeIcon icon={faCaretDown} className="ml-2" />
+      </div>
+      {isTokenModalOpened && <TokenModal onModalClosed={toggleAccountModal} />}
+    </div>
+  )
+}
+
+function TokenModal({ onModalClosed }: { onModalClosed: () => void }) {
+  const { provider } = useProviderContext()
+  const { importToken } = useAction()
+  const account = useAccount()
+
+  const [tokenAddress, setTokenAddress] = useState<HexString>("")
+  const [tokenSymbol, setTokenSymbol] = useState<string>("")
+  const [tokenDecimals, setTokenDecimals] = useState<number>(0)
+  const [invalidTokenAddress, setInvalidTokenAddress] = useState<boolean>(true)
+  const [invalidTokenSymbol, setInvalidTokenSymbol] = useState<boolean>(true)
+
+  const handleTokenAddressChange = async (
+    event: ChangeEvent<HTMLInputElement>
+  ) => {
+    const inputTokenAddress = event.target.value
+    setTokenAddress(inputTokenAddress)
+    const erc20 = getErc20Contract(inputTokenAddress, provider)
+
+    try {
+      console.log(`${ethers.getAddress(inputTokenAddress)}`)
+      const symbol: string = await erc20.symbol()
+      const decimals: number = ethers.toNumber(await erc20.decimals())
+      setInvalidTokenAddress(false)
+      setInvalidTokenSymbol(false)
+      setTokenSymbol(symbol)
+      setTokenDecimals(decimals)
+    } catch (error) {
+      console.warn(`[Popup][tokens] Invalid token address: ${error}`)
+      setInvalidTokenAddress(true)
+      setInvalidTokenSymbol(true)
+      setTokenSymbol("")
+      setTokenDecimals(0)
+    }
+  }
+
+  const handleTokenSymbolChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const inputTokenSymbol = event.target.value
+    setTokenSymbol(inputTokenSymbol)
+    setInvalidTokenSymbol(inputTokenSymbol.length === 0)
+  }
+
+  const onTokenImported = async (event: FormEvent<HTMLFormElement>) => {
+    // Prevents the default behavior of the event,
+    // allowing for custom handling of the event action.
+    event.preventDefault()
+
+    let balance: BigNumberish
```

Comment:
nit: Init it with 0 and remove line 106

```suggestion
    let balance: BigNumberish = 0
```